src/main/java/com/microsoft/graph/requests/extensions/AppRoleAssignmentReferenceRequest.java

```diff
@@ -15,6 +15,8 @@
 import com.microsoft.graph.http.BaseRequest;
 import com.microsoft.graph.http.HttpMethod;
 import com.microsoft.graph.core.IBaseClient;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonObject;
 
 // **NOTE** This file was generated by a tool and any changes will be overwritten.
 
```

Comment:
I think that we don't need to generate this file at all given that all instances of AppRoleAssignments are `ContainsTarget=true`.

Reply:
thanks for catching that, it was on my todo list to review it but I somehow forgot...
You'll see that in the recent commits I pushed, the return type of propertyAsSomething methods (cast) changed. That's because the initial iteration of the odata cast implementation was wrongly introducing a lot of reference things in the mix.
Because this could be a potentially breaking change to people who might have already started using casting, I'll rebase on v3. Hold any review for now please.